.github/workflows/publish-charts.yaml

```diff
@@ -0,0 +1,85 @@
+name: publish-charts
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Repository from where we want to build Charts'
+        required: true
+      repo_ref:
+        description: 'The branch/tag that will be used as new version for charts. Usually tag ref'
+        required: true
+      repo_chart:
+        description: 'Object with source path, destination path and chart name'
+        required: true
+
+env:
+  REPO: ${{ github.event.inputs.repo }}
+  REPO_REF: ${{ github.event.inputs.repo_ref }}
+
+jobs:
+  publish-charts:
+    runs-on: ubuntu-latest
+    name: Publish charts
+    continue-on-error: true
+    strategy:
+      max-parallel: 1
+      matrix: ${{ fromJson(github.event.inputs.repo_chart) }}
+    steps:
+
+      - id: checkout-gh-pages
+        uses: actions/checkout@v2 # Checkout Charts - gh-pages
+        with:
+          ref: 'gh-pages'
+          fetch-depth: 0
+
+      - id: checkout-helm-src
+        uses: actions/checkout@v2 # Checkout Helm source code
+        with:
+          repository: ${{ env.REPO }}
+          ref: ${{ env.REPO_REF }}
+          path: ${{ env.REPO }}
+
+      - id: chart-version
+        name: 'Set version - ${{ env.REPO_REF }}'
+        if: ${{ steps.checkout-gh-pages.outcome == 'success' && steps.checkout-helm-src.outcome == 'success' }}
+        run: |
+          yq -i '.version = "${{ env.REPO_REF }}"' ${{ matrix.src_path }}Chart.yaml
+          yq -i '.version = "${{ env.REPO_REF }}"' ${{ matrix.src_path }}values.yaml
+          echo "Chart version: ${{ env.REPO_REF }}."
+        working-directory: ./${{ env.REPO }}
+
+      - id: check
+        name: 'Check if chart already exist'
+        if: ${{ steps.chart-version.outcome == 'success' }}
+        run: |
+          FILE=$(grep name ${{ env.REPO }}/${{ matrix.src_path }}Chart.yaml | awk '{print $2}')
+          if [[ -f "${{ matrix.dst_path }}$FILE-${{ env.REPO_REF }}.tgz" ]]; then 
+            echo "Chart already exists." 
+            echo "${{ matrix.dst_path }}$FILE-${{ env.REPO_REF }}.tgz"
+            ls -larth ${{ matrix.dst_path }}
+            exit 1
+          else
+            echo "File doesn't exist, continue with packaging"
+          fi
+
+      - id: package
+        name: Package ${{ matrix.chart }}-${{ env.REPO_REF }}.tgz
```

Comment:
Can we add `${{ matrix.chart }}-${{ env.REPO_REF }}.tgz` as an output of some of the previous steps, and just read the output, so that we don't expand this same expression multiple times, and contain the logic of compiling the path and the name of the file in one place?

Reply:
that's good idea! i've pushed the change please check if that is what you ment

Reply:
that's it :)